ivi/rohdeschwarz/rohdeschwarzBaseScope.py

```diff
@@ -0,0 +1,1057 @@
+        'noise_reject_dc': ('dc', 0, 1),
+        'noise_reject_ac': ('ac', 0, 1)}
+PolarityMapping = {'positive': 'pos',
+        'negative': 'neg'}
+GlitchConditionMapping = {'less_than': 'less',
+        'greater_than': 'gre'}
+WidthConditionMapping = {'within': 'rang'}
+SampleModeMapping = {'real_time': 'rtim',
+        'equivalent_time': 'etim',
+        'segmented': 'segm'}
+SlopeMapping = {
+        'positive': 'pos',
+        'negative': 'neg',
+        'either': 'eith'}
+MeasurementFunctionMapping = {
+        'rise_time': 'risetime',
+        'fall_time': 'falltime',
+        'frequency': 'frequency',
+        'period': 'period',
+        'voltage_rms': 'vrms display',
+        'voltage_peak_to_peak': 'vpp',
+        'voltage_max': 'vmax',
+        'voltage_min': 'vmin',
+        'voltage_high': 'vtop',
+        'voltage_low': 'vbase',
+        'voltage_average': 'vaverage display',
+        'width_negative': 'nwidth',
+        'width_positive': 'pwidth',
+        'duty_cycle_positive': 'dutycycle',
+        'amplitude': 'vamplitude',
+        'voltage_cycle_rms': 'vrms cycle',
+        'voltage_cycle_average': 'vaverage cycle',
+        'overshoot': 'overshoot',
+        'preshoot': 'preshoot',
+        'ratio': 'vratio',
+        'phase': 'phase',
+        'delay': 'delay'}
+MeasurementFunctionMappingDigital = {
+        'rise_time': 'risetime',
+        'fall_time': 'falltime',
+        'frequency': 'frequency',
+        'period': 'period',
+        'width_negative': 'nwidth',
+        'width_positive': 'pwidth',
+        'duty_cycle_positive': 'dutycycle'}
+MeasurementStatusMapping = {
+        'complete': 'comp',
+        'in_progress': 'stop',
+        'running': 'run',
+        'break': 'bre'}
+ScreenshotImageFormatMapping = {
+        'bmp': 'bmp',
+        'png': 'png'}
+TimebaseModeMapping = {
+        'main': 'main',
+        'window': 'wind',
+        'xy': 'xy',
+        'roll': 'roll'}
+TimebaseReferenceMapping = {
+        'left': 8.33,
+        'center': 50.0,
+        'right': 91.67}
+
+class rohdeschwarzBaseScope(scpi.common.IdnCommand, scpi.common.ErrorQuery, scpi.common.Reset,
+                            scope.Base, scope.ContinuousAcquisition, scope.TriggerModifier, scope.Interpolation,
+                            scope.WaveformMeasurement,
+                            extra.common.Screenshot,
+                            ivi.Driver):
+    "Rohde&Schwarz generic IVI oscilloscope driver"
+    
+    def __init__(self, *args, **kwargs):
+        self.__dict__.setdefault('_instrument_id', '')
+        self._analog_channel_name = list()
+        self._analog_channel_count = 4
+        self._digital_channel_name = list()
+        self._digital_channel_count = 16
+        self._channel_count = self._analog_channel_count + self._digital_channel_count
+        self._channel_label = list()
+        
+        self._vertical_divisions = 10
+        
+        # self._channel_probe_skew = list()
```

Comment:
Any reason while we want to keep commented-out code? In that case please motivate the reason with a comment. Otherwise please remove the commented-out code.